File: wx_mini_program_golang/db/model/message.go
package model

import "time"

const TableNameMessage = "message"

type Message struct {
	ID              int64     `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	FamilyID        int64     `gorm:"column:family_id" json:"familyId"`
	UserID          string    `gorm:"column:user_id" json:"userId"`
	UserName        string    `gorm:"column:user_name" json:"userName"`
	UserAvatar      string    `gorm:"column:user_avatar" json:"userAvatar"`
	Content         string    `gorm:"column:content" json:"content"`
	ParentID        *int64    `gorm:"column:parent_id" json:"parentId"`
	ReplyToUserID   string    `gorm:"column:reply_to_user_id" json:"replyToUserId"`
	ReplyToUserName string    `gorm:"column:reply_to_user_name" json:"replyToUserName"`
	CreateTime      time.Time `gorm:"column:create_time;autoCreateTime" json:"createTime"`
}

func (*Message) TableName() string {
	return TableNameMessage
}

File: miniprogram/api/message.ts
import { request } from '../utils/request';
import type { Message } from '../models/index';

export async function addMessage(content: string, parentId: string | null = null, replyToUserId?: string, replyToUserName?: string): Promise<Message> {
  const data: any = { content };
  if (parentId) data.parentId = parseInt(parentId);
  if (replyToUserId) data.replyToUserId = replyToUserId;
  if (replyToUserName) data.replyToUserName = replyToUserName;

  const res = await request<any>('/message/add', 'POST', data);
  return mapMessage(res);
}

export async function getMessages(): Promise<Message[]> {
  const res = await request<any[]>('/message/list', 'GET');
  return res.map(mapMessage);
}

function mapMessage(m: any): Message {
  return {
    id: String(m.id),
    familyId: String(m.family_id ?? m.familyId),
    userId: m.user_id ?? m.userId,
    userName: m.user_name ?? m.userName,
    userAvatar: m.user_avatar ?? m.userAvatar,
    content: m.content,
    createTime: new Date(m.create_time ?? m.createTime).getTime(),
    parentId: (m.parent_id ?? m.parentId) ? String(m.parent_id ?? m.parentId) : null,
    replyToUserId: m.reply_to_user_id ?? m.replyToUserId,
    replyToUserName: m.reply_to_user_name ?? m.replyToUserName,
    replies: m.replies ? m.replies.map(mapMessage) : []
  };
}

export async function deleteMessage(msgId: string): Promise<void> {
  await request(`/message/delete?id=${msgId}`, 'POST');
}
